config/config.go

```diff
@@ -231,48 +237,79 @@ func ParseConfigFile(configPath string, terragruntOptions *options.TerragruntOpt
 }
 
 // Parse the Terragrunt config contained in the given string and merge it with the given include config (if any)
-func ParseConfigString(configString string, terragruntOptions *options.TerragruntOptions, include *IncludeConfig, configPath string) (*TerragruntConfig, error) {
-	terragruntConfigFile, err := parseConfigStringAsTerragruntConfigFile(configString, configPath, terragruntOptions, include)
+func ParseConfigString(configString string, terragruntOptions *options.TerragruntOptions, includeFromChild *IncludeConfig, filename string) (*TerragruntConfig, error) {
+	file, err := parseHcl(configString, filename)
 	if err != nil {
 		return nil, err
 	}
-	if terragruntConfigFile == nil {
-		return nil, errors.WithStackTrace(CouldNotResolveTerragruntConfigInFile(configPath))
-	}
 
-	config, err := convertToTerragruntConfig(terragruntConfigFile, configPath, terragruntOptions)
+	terragruntInclude, err := decodeAsTerragruntInclude(file, filename, terragruntOptions)
```

Comment:
Please add comments explaining why we do this two-step parse process

Reply:
In fact, now that the parsing is more complicated, each of the subsequent steps could use some explanation

Reply:
The new comments are great, thanks!